torba/hash.py

```diff
@@ -106,14 +106,17 @@ def hex_str_to_hash(x):
     return reversed(unhexlify(x))
 
 
-def aes_encrypt(secret, value):
+def aes_encrypt(secret, value, init_vector=None):
+    if init_vector:
+        assert len(init_vector) == 16
+    else:
+        init_vector = os.urandom(16)
     key = double_sha256(secret)
-    init_vector = os.urandom(16)
     encryptor = Cipher(AES(key), modes.CBC(init_vector), default_backend()).encryptor()
     padder = PKCS7(AES.block_size).padder()
     padded_data = padder.update(value) + padder.finalize()
-    encrypted_data2 = encryptor.update(padded_data) + encryptor.finalize()
-    return base64.b64encode(encrypted_data2)
+    encrypted_data = encryptor.update(padded_data) + encryptor.finalize()
+    return base64.b64encode(init_vector + encrypted_data)
```

Comment:
Why does the `init_vector` need to be configurable? Is there a place where a custom `init_vector` is passed? If we're not overriding it somewhere already I'd rather hold off on adding this customization until we actually need it.

Reply:
I haven't updated the (currently skipped) wallet encryption tests - it's needed for those